src/runtime_src/core/common/api/xrt_kernel.cpp

```diff
@@ -1481,7 +1481,20 @@ class kernel_impl : public std::enable_shared_from_this<kernel_impl>
       kcmd->opcode = (protocol == control_type::fa) ? ERT_START_FA : ERT_START_CU;
       break;
     case kernel_type::dpu :
-      kcmd->opcode = (m_module ? ERT_START_DPU : ERT_START_CU);
+      if(m_module)
+      {
+        // TBD: this is temporary until ELF has save/restore sections
```

Comment:
We can hard coding the save/restore transaction-buffer content so we can proceed with what we have.

// hexdump - v - e '6/4 "0x%08x, "' - e '"\n"' . / patched_preempt_restore_1col.bin
static uint32_t preempt_restore_1col[] = {
0x00000000, 0x0000010c, 0x00000000, 0x000001a8, 0x06040100, 0x00000104,
0x0000000d, 0x000001a0, 0x00000100, 0x00000000, 0x001b0000, 0x00000000,
0x80000007, 0x00000018, 0x001c0100, 0x00000000, 0x001b011c, 0x00000000,
0x80000000, 0x00000018, 0x00300000, 0x00000000, 0x0003f030, 0x00000000,
0x80000005, 0x00000018, 0x00140000, 0x00000000, 0x0003f114, 0x00000000,
0x80000000, 0x00000018, 0x00000003, 0x00000000, 0x0001f000, 0x00000000,
0x00000400, 0x00000c00, 0x00000020, 0x00000000, 0x00000001, 0x00000000,
0x0001d000, 0x00000030, 0x00020000, 0x00000000, 0x00000000, 0x00000000,
0x80000000, 0x00000000, 0x00000000, 0x02000000, 0x00000101, 0x00000000,
0x001a0000, 0x00000030, 0x00010000, 0x00020000, 0x00000000, 0x00000000,
0x00000000, 0x00000000, 0x007effff, 0x80000000, 0x00000081, 0x00000030,
0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x0001d004, 0x00000000,
0x00000001, 0x00000000, 0x00000000, 0x00000000, 0x00140000, 0x00000000,
0x0001d214, 0x00000000, 0x00000000, 0x00000018, 0x00040100, 0x00000000,
0x001a0604, 0x00000000, 0x00010000, 0x00000018, 0x00100000, 0x00000000,
0x0001d210, 0x00000000, 0x00000001, 0x00000018, 0x00000100, 0x00000000,
0x001a0600, 0x00000000, 0x00000001, 0x00000018, 0x00600104, 0x00000000,
0x001a0660, 0x00000000, 0x00000000, 0x0078003c, 0x00000020, 0x00000000,
};
constexpr size_t preempt_restore_1col_size = sizeof(preempt_restore_1col);
 
// hexdump -v -e '6/4 "0x%08x, "' -e '"\n"' ./patched_preempt_save_1col.bin
static uint32_t preempt_save_1col[] = {
0x00000000, 0x0000010c, 0x00000000, 0x000001a8, 0x06040100, 0x00000104,
0x0000000d, 0x000001a0, 0x001c0100, 0x00000000, 0x001b001c, 0x00000000,
0x80000000, 0x00000018, 0x00000100, 0x00000000, 0x001b0100, 0x00000000,
0x80000000, 0x00000018, 0x00100000, 0x00000000, 0x0003f010, 0x00000000,
0x8000000e, 0x00000018, 0x00380000, 0x00000000, 0x0003f138, 0x00000000,
0x80000000, 0x00000018, 0x00040003, 0x00000000, 0x0001f004, 0x00000000,
0x00000010, 0x00000030, 0x00000020, 0x00000000, 0x00000101, 0x00000000,
0x001a0000, 0x00000030, 0x00010000, 0x00020000, 0x00000000, 0x00000000,
0x00000000, 0x00000000, 0x007effff, 0x80000000, 0x00000001, 0x00000000,
0x0001d000, 0x00000030, 0x00020000, 0x00000000, 0x00000000, 0x00000000,
0x80000000, 0x00000000, 0x00000000, 0x02000000, 0x00000081, 0x00000030,
0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x0001d004, 0x00000000,
0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00340100, 0x00000000,
0x001a0634, 0x00000000, 0x00010000, 0x00000018, 0x00040000, 0x00000000,
0x0001d204, 0x00000000, 0x00000000, 0x00000018, 0x00300100, 0x00000000,
0x001a0630, 0x00000000, 0x00000001, 0x00000018, 0x00000000, 0x00000000,
0x0001d200, 0x00000000, 0x00000001, 0x00000018, 0x00200004, 0x00000000,
0x0001d220, 0x00000000, 0x00000000, 0x0078003c, 0x00000020, 0x00000000
};